pkg/controller/bookkeepercluster/bookkeepercluster_controller.go

```diff
@@ -234,31 +235,47 @@ func (r *ReconcileBookkeeperCluster) syncBookieSize(bk *bookkeeperv1alpha1.Bookk
 func (r *ReconcileBookkeeperCluster) reconcileFinalizers(bk *bookkeeperv1alpha1.BookkeeperCluster) (err error) {
 	if bk.DeletionTimestamp.IsZero() {
 		if !util.ContainsString(bk.ObjectMeta.Finalizers, util.ZkFinalizer) {
+			// appending name of pravega cluster to the finalizers
+			configMap := &corev1.ConfigMap{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: strings.TrimSpace(bk.Spec.EnvVars), Namespace: bk.Namespace}, configMap)
```

Comment:
bkSpecEnvVars may be empty...in which case the Get call with always fail....it may help to add a empty string check for `bk.Spec.EnvVars`

Reply:
done